### PR TITLE
chore: migrate to boxo

### PR DESCRIPTION
**Pull request by Jorropo**
This migrates everything except the `go-car` librairy: https://github.com/ipfs/boxo/issues/218#issuecomment-1529922103

### CODE REVIEW

**Comment by Jorropo** [comment]
DO NOT MERGE UNTIL https://github.com/ipfs/boxo/issues/291 is fixed (and this PR is bumped to a version of boxo or go-car where this is fixed)

**Review by ZenGround0 — APPROVED** [review]

**Comment by ZenGround0** [comment]
I don't have authorization here but LGTM

**Review by dirkmc — APPROVED** [review]
Approved, conditional on updating all the dependencies to release versions.

**Pull request merged**
